components/core/src/clp/streaming_archive/writer/Archive.hpp

```diff
@@ -279,6 +279,11 @@ class Archive {
      */
     void update_metadata();
 
+    /**
+     * Updates metadata in the global metadata database.
```

Comment:
ditto comment for deleting m_files_written.

You might also need to mention about exceptions if m_global_metadata_db's api throws any.